src/pybind/mgr/dashboard/controllers/grafana.py

```diff
@@ -17,20 +17,25 @@
 @ControllerDoc("Grafana Management API", "Grafana")
 class Grafana(BaseController):
 
+    @staticmethod
+    def _grafana_frontend_url():
+        # type: () -> str
+        return (str(Settings.GRAFANA_FRONTEND_API_URL) or str(Settings.GRAFANA_API_URL)).rstrip('/')
+
     @Endpoint()
     @ReadPermission
     @EndpointDoc("List Grafana URL Instance",
                  responses={200: URL_SCHEMA})
     def url(self):
-        response = {'instance': Settings.GRAFANA_API_URL}
+        response = {'instance': self._grafana_frontend_url()}
```

Comment:
```suggestion
        response = {'instance': str(Settings.GRAFANA_FRONTEND_API_URL or Settings.GRAFANA_API_URL).rstrip('/')}
```
Or, IMHO, even better:
```suggestion
        response = {'instance': (mgr.get_module_option('GRAFANA_FRONTEND_API_URL') or mgr.get_module_option('GRAFANA_API_URL')).rstrip('/')}
```


`self._grafana_frontend_url()` is a thin wrapper around `Settings.GRAFANA_...` which is in turn a thin wrapper around `mgr.get_module_option('GRAFANA_...')`, so by calling a static method in the Grafana controller we're triggering a read to ceph-mgr config option cache... So instead of encapsulating complexity (which there is not) this is just creating a false sense of where information belongs to...